nexus/types/src/external_api/params.rs

```diff
@@ -297,6 +303,50 @@ pub struct SiloCreate {
         BTreeMap<shared::SiloRole, BTreeSet<shared::FleetRole>>,
 }
 
+/// The amount of provisionable resources for a Silo
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct SiloQuotasCreate {
+    /// The amount of virtual CPUs available for running instances in the Silo
+    pub cpus: i64,
+    /// The amount of RAM (in bytes) available for running instances in the Silo
+    pub memory: ByteCount,
+    /// The amount of storage (in bytes) available for disks or snapshots
+    pub storage: ByteCount,
+}
+
+impl SiloQuotasCreate {
+    /// All quotas set to 0
+    pub fn empty() -> Self {
+        Self {
+            cpus: 0,
+            memory: ByteCount::from(0),
+            storage: ByteCount::from(0),
+        }
+    }
+
+    /// 30% of CPUs and memory reserved for internal usage.
+    /// Storage calculated at (total / 3.5) to account for redundancy / bookkeeping.
+    pub fn half_rack() -> Self {
+        Self {
+            cpus: 90,
+            memory: ByteCount::from_gibibytes_u32(708),
+            storage: ByteCount::from_gibibytes_u32(850),
+        }
+    }
```

Comment:
Just remembered this is only the resources for a single sled. Need to multiply this by... 16?

Reply:
It's basically a shortcut to set a decent top line quota that (will when I fix it) equate out to the calculated capacity of a half rack. Right now it's basically used for tests and other places we need to allocate some amount of resources that can't be 0.

Reply:
I've updated this function to be called `from_sled_count` and to take in a number of sleds. So the usage is like `SiloQuotaCreate::from_sled_count(16)` for a half rack. 

This function is mostly a helper/placeholder. All of the places we have current (including the default silo) are largely for testing. I don't think it's worth doing anything fancy for the default silo b/c we should ultimately be removing it. Further, this number isn't really all that accurate. To do this correctly, you'd want to query all the sleds, get there hardware based numbers, and use those to calculate the totals with the slush factors. We will do that at some point but even then these usages might not be worth updating. 

Regardless, this doesn't need to be smart right now and it's here b/c to aid in creating quotas with non-empty silos.